backend/api/models.py

```diff
@@ -33,6 +36,28 @@ class FullMoodleSettings(BaseMoodleSettings):
     id: int
 
 
+class BaseCourseBuildSettings(BaseModel):
+    instructor_firstname: str
+    instructor_lastname: str
+    instructor_email: str
+    school_district: int
```

Comment:
This should probably be the name `str` and not the database `int`. Or, maybe we include both if we want, but in the end when returning course build information we'll definitely want a friendly string to display in the UI. I'm also okay with us taking just the string as input since if it is invalid, we'll fail on `INSERT` due to the FK relationship.

Reply:
If `school_district` in the above input model is updated to a `str`, does that mean that I would need to use the `school_district` name to get the `id` associated with that name in the `school_distrct` table? We would still need the `id` in order to add the course build data to the `course_build` table.

Reply:
Yes, you'll have to create the FK relationship. I guess my comment here is there are two interfaces and we should think of them both separately:

* API - What is the "model" for a FE client to use?
* DB - The model that is enforced by the schema

Sometimes (as I think in this case), there is a bit of separation between the two where business logic needs to be added. But, if we don't actively think about this design, it's too easy to just "pipe through" and then the abstractions used on one interface no longer make sense.

Reply:
That's a great way to frame the two interfaces. I was not actively making the distinction between the two. 

My initial thought is to create a function in `database.py` that will query `SchoolDistrict` and filter on the name provided by the user input. Assuming the district exists, this should provide us with the `id` in `SchoolDistrict` to then be used as part of our insert of course build data into the `course_build` table. Our API interface would receive from and return to the user `school_district` as a `str`.